low-level/sensors/impl/generic/cpu_fault_sensor.py

```diff
@@ -244,14 +244,20 @@ def _create_json_message(self, cpu, alert_type):
         self.fill_specific_info()
         alert_specific_info = self.specific_info
 
+        if alert_type == "fault":
+            description = "Fault is detected for cpu core"
```

Comment:
ANy idea what fault detected for cpu core?

Same applies for faul_resolved

Reply:
@cdeshmukh , here the current list of CPUs that are online is found.  Previous list of CPUs is stored in store.
- fault is detected if a previously present CPU is not found in the current list. 
- fault _resolved is created if a new CPU is found in the current list that was not present earlier AND an outstanding fault alert is present for that CPU.

Reply:
@cdeshmukh  updated description, please check.

Reply:
Well updated description not good enough.

>fault is detected if a previously present CPU is not found in the current list.

current list is currently detected CPU list?

>AND an outstanding fault alert is present for that CPU.

We give fault_resolved when fault exists on newly detected CPU? seems confusing

Reply:
what should I mention in description, can you please suggest @cdeshmukh  @ArchanaLimaye

Reply:
Well am not sure either. Also, if CPU is not detected, wouldn't it missing alert?
@saurabhkhanvilkar possible to get a word from Archana ?

Reply:
@cdeshmukh please find replies inline

> 
> > fault is detected if a previously present CPU is not found in the current list.
> 
> current list is currently detected CPU list?

Yes 

[  >>  if a new CPU is found in the current list that was not present earlier **AND** an outstanding fault alert is present for that CPU] 

"fault_resolved" is sent only if a "fault" alert was raised for that CPU earlier.

Reply:
> Well am not sure either. Also, if CPU is not detected, wouldn't it missing alert?
@cdeshmukh  not sure what you mean here, 

The algorithm is :
- On the very first bootup of SSPL, collect the list of all active/online CPUs and store it in file_store
- On every SSPL power cycle, get the list of all online CPUs (current_list). Also get the stored list from file_store (prev_list)
- compare the two lists : current_list and prev_list
          -  If  CPU is present in prev_list but not present in current_list : raise "fault". Add the CPU to list of faulty 
              CPUs  (faulty_cpus)  
          - if CPU is present in current_list but not in prev_list:
                         -  if CPU is present in "faulty_cpus", raise fault_resolved alert, remove the CPU from "faulty_CPUs"
                         - else do nothing
          - store the current_list as prev_list oin file_store. This will be used for next comparison cycle (on next SSPL reboot)   



Regarding the alert description: 
The description can be something like:

for FAULT CASE:   "faulty CPU detected,  {cpuname} is offline "
for FAULT_RESOLVED case:  "fault resolved for CPU, {cpuname} is now online "

Reply:
updated description. as per comment.